### PR TITLE
Support check.test requests (CMC-1506)

### PR DESCRIPTION
**Pull request by itzg**
### TODO

- [ ] Testing
- [ ] Integrate with final request/protocol definition

### CODE REVIEW

File: poller/session.go [hunk 1]
```diff
@@ -440,6 +443,32 @@ func (s *EleSession) handlePollerCommit(f *protocol.FrameMsg) {
 	s.prepDetails.commit()
 }
 
+func (s *EleSession) handleCheckTest(f *protocol.FrameMsg) {
+	req := protocol.DecodePollerCheckTestRequest(f)
+
+	newCheck, err := check.NewCheckParsed(s.ctx, *req.Params)
+	if err != nil {
+		resp := protocol.NewErrorResponse(f, 1, "Unable to interpret the given check for testing")
```
Comment (itzg) [inline poller/session.go]: @rphillips is there a strategy from virgo agent to define appropriate error code values?

Comment (rphillips) [inline poller/session.go]: No, not really.

https://github.com/virgo-agent-toolkit/rackspace-monitoring-agent/blob/master/check/init.lua#L85

File: poller/session_test.go [hunk 2]
```diff
@@ -573,11 +573,13 @@ func TestEleSession_PollerPrepare(t *testing.T) {
 			origTimestamper := installDeterministicTimestamper(1000, 2000)
 			defer utils.InstallAlternateTimestampFunc(origTimestamper)
 
-			cfg := config.NewConfig("1-2-3", false, nil)
+			cfg := config.NewConfig(tt.name, false, nil)
 			es := poller.NewSession(context.Background(), eleConn, reconciler, cfg)
+			defer func() { time.Sleep(5 * time.Millisecond) }()
```
Comment (itzg) [inline poller/session_test.go]: I was seeing some sporadic unit test failures here...but pushing so I can switch to working on ping/ICMP

File: protocol/messages.go [hunk 3]
```diff
@@ -172,7 +174,11 @@ type MetricTVU struct {
 	Unit  string `json:"u"`
 }
 
-type MetricsPostRequestParams struct {
+func (m *MetricTVU) String() string {
+	return fmt.Sprintf("{t=%v, v=%v, u=%v}", m.Type, m.Value, m.Unit)
```
Comment (rphillips) [inline protocol/messages.go]: nice

**Pull request merged**